contrib/checkbox-provider-ce-oem/bin/serial_test.py

```diff
@@ -50,92 +51,103 @@ def init_logger():
     return root_logger
 
 
-def str_generator(size):
-    chars = []
-    chars.extend(string.ascii_uppercase)
-    chars.extend(string.ascii_lowercase)
-    chars.extend(string.digits)
-    chars.extend(string.punctuation)
-
-    return "".join(random.choices(chars, k=size))
-
-
-def serial_init(device, **kwargs):
-    ser = serial.Serial(
-        device,
-        baudrate=kwargs.get("baudrate", 115200),
-        bytesize=kwargs.get("bytesize", 8),
-        parity=kwargs.get("parity", "N"),
-        stopbits=kwargs.get("stopbits", 1),
-        timeout=1,
-        write_timeout=1,
-        xonxoff=True
-    )
-    return ser
-
-
-def sender(ser, test_str):
-    try:
-        ser.write(test_str.encode("utf-8"))
-        logging.info("Sent: {}".format(test_str))
-    except Exception:
-        logging.error("Not able to send data!")
-
-
-def receiver(ser):
-    """
-    If trying to receive string between two different protocols
-    (e.g. RS485 with RS232). Then it will receive the string
-    that is not able to decode. So we can handle that kind of
-    an exception to filter out the string from the different protocols.
-    """
-    rcv = ""
-    try:
-        rcv = ser.readline().decode("utf-8")
-        if rcv:
-            logging.info("Received: {}".format(rcv))
-    except ValueError:
-        logging.error("Received unmanageable string format")
-        rcv = "Error format"
-    return rcv
-
-
-def server_mode(ser):
+class Serial:
+    def __init__(
+        self,
+        node,
+        type,
+        group: list = [],
+        baudrate: int = 115200,
+        bytesize: int = 8,
+        parity: str = "N",
+        stopbits: int = 1,
+        timeout: int = 1,
+    ) -> None:
+        self.node = node
+        self.type = type
+        self.baudrate = baudrate
+        self.bytesize = bytesize
+        self.parity = parity
+        self.stopbits = stopbits
+        self.timeout = timeout
+        self.ser = self.serial_init(node)
+        self.group = []
+        for ser in group:
+            self.group.append(self.serial_init(ser))
+
+    def serial_init(self, node: str) -> serial.Serial:
+        """Create a serial.Serial object based on the class variables"""
+        ser = serial.Serial(
+            node,
+            baudrate=self.baudrate,
+            bytesize=self.bytesize,
+            parity=self.parity,
+            stopbits=self.stopbits,
+            timeout=1,
+        )
+        if self.type == "RS485":
+            ser.rs485_mode = serial.rs485.RS485Settings()
+        ser.reset_input_buffer()
+        ser.reset_output_buffer()
+        return ser
+
+    def send(self, data: bytes) -> None:
+        try:
+            self.ser.rts = True
+            self.ser.write(data)
+            self.ser.rts = False
```

Comment:
Could you help me to understand why you switch the `rts` value before and after `write`?

Reply:
Actually we don't need this.